cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/artifact/ArtifactStore.java

```diff
@@ -125,7 +128,8 @@
   private static final Id.DatasetInstance META_ID =
     Id.DatasetInstance.from(Constants.SYSTEM_NAMESPACE, "artifact.meta");
 
-  private final NamespacedLocationFactory locationFactory;
+  private final LocationFactory locationFactory;
+  private final NamespacedLocationFactory namespacedLocationFactory;
```

Comment:
The naming for the `NamespacedLocationFactory` is really bad.. It is not a `LocationFactory` at all. Anyway not related to this PR, but that needs to be fixed.